### PR TITLE
Add GitHub workflows for automating Release creation

### PR DESCRIPTION
**Pull request by andrewlock**
This creates 4 new GitHub Actions workflows:
* Auto add PR to `vNext` milestone
* Create version bump PR
* Auto tag version bump commit
* Create draft release from latest tag

> This PR is currently rebased on top of #1605 as it requires changes in there, but it is only the final commit.
> Also, as GitHub actions are hard to test, I've been testing this in a private repo, and seems to work ok to me!

## Auto add PR to vNext milestone

To make later stages of the release workflow easier, this will automatically run when a pull request is closed, and add it to a `vNext` milestone. Only _merged_ PRs that will be included in the next release should be added to the `vNext` milestone.

## Create version bump PR

When we're ready to do a release, the first step is to do a "version bump" commit. This workflow can be _manually_ triggered from the GitHub actions page, where you provide the next version number, and whether or not it's a prerelease.

image

The action first updates the changelog, using the titles of all PRs included in the `vNext` milestone, using the existing format.

Next it does the version bump, updates integrations.json, and syncs the MSI contents.

After that's complete it renames the `vNext` milestone to the appropriate value, e.g. `1.28.2-prerelease`.

Finally, it creates a branch called `version-bump-1.28.2-prerelease` (for example), commits the changed files to it, pushes, and opens a PR.

Users can then review the PR, make changes if required, and merge the PR.

## Auto tag version bump commit

When the `[Version Bump]` commit above is merged to master, a new workflow is kicked off, that automatically pushes a new tag, `v1.28.2-prerelease` to the repo. This also creates the benchmark branch (as the previous worklfow was doing)

 ## Create draft release from latest tag

 The final workflow creates the draft release from the latest tag assets. This assumes that you've already done a version bump PR (it doesn't _require_ that it's the auto-bump, it just needs to have an entry in the changelog, and assets in Azure Devops).

 The workflow starts by downloading the assets for the `v1.28.2-prerelease` tag build from Azure Devops. We publish a "well known" artifact that it uses to pull everything down, warning/erroring if it can't find an expected build, if the build isn't complete, or if it failed to produce any assets.

 Once download and extracted, it extracts the release notes from the changelog, and uses these to create a draft release. The release notes include warning/reminders about uploading the nuget packages to nuget, and about attaching the signed MSI artifacts

 Final publishing is up to a user, there's no automation around that

### CODE REVIEW

**Review by lucaspimentel — APPROVED** [review]
Fabulous

File: .github/workflows/auto_add_vnext_milestone_to_pr.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+on:
+  pull_request:
+    branches:
+      - master
+      - main
+    types: [closed]
+
+jobs:
+  add_to_milestone:
+    if: github.event.pull_request.merged == true && startsWith(github.event.pull_request.title, '[Version Bump]') == false
```
[the remark this answers is not in the document]
Comment (andrewlock) [reply .github/workflows/auto_add_vnext_milestone_to_pr.yml:12]: No, we need to keep the condition, but version bump PRs are still added to the correct milestone.

As part of the version bump PR workflow, we rename the `vNext` milestone to the appropriate value (e.g. `1.28.2`), and set the milestone on the PR. This workflow runs when the PR closes, and would set the milestone to `vNext`, which would be wrong at this point :)

[the remark this answers is not in the document]
Comment (lucaspimentel) [reply .github/workflows/auto_add_vnext_milestone_to_pr.yml:12]: I see, thanks!

File: .github/workflows/create_version_bump_pr.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          branch: "version-bump-${{steps.versions.outputs.full_version}}"
+          commit-message: "[Version Bump] ${{steps.versions.outputs.full_version}}"
+          delete-branch: true
+          title: "[Version Bump] ${{steps.versions.outputs.full_version}}"
+          milestone: ${{steps.rename.outputs.milestone}}"
+          reviewers: "DataDog/apm-dotnet"
+          body: |
+            The following files should have been modified. Confirm that each file was updated correctly.
```
Comment (zacharycmontoya) [inline .github/workflows/create_version_bump_pr.yml:58]: 😍

**Review by zacharycmontoya — APPROVED** [review]
This is awesome!

**Pull request merged**